### ISSUE TITLE
[feature] Putting code selection directly into handlebars output

### ISSUE THREAD

**Issue by bluenick2k17** [d-koppenhagen/vscode-code-review#35]
# 🧩 Feature
Removing the HTML script tag from the handlebars output and inserting direct code snippets
## Description

My use case for this tool is to try and copy the HTML from the Handlebars generated report, and insert it into an HTML comments box to give formatted feedback to students. However, this comments box does not support script tags in them (to be fair, for good reason), and so I wanted to see the feasibility of trying to re-place the functionality in the `<script>` tag elsewhere.

## Solutions

I see there being two viable options:
1. Doing any necessary character escaping on the backend, and passing in the full code block to Handlebars (and not the base64 encoded version)
2. Using a Handlebars helper to display the code blocks (converting from base64 and escaping) instead of using a script tag in the HTML.

I can't fully say what the first one would look like, but I can try to provide some context for the second one. I believe you would be able to register a helper before `src/export-factory.ts:39` like so:

```javascript
Handlebars.registerHelper('codeBlock', function (code) {
      const base64DecodedContent = atob(code);
      const codeBlock = base64DecodedContent.replace(/</g,"&lt;");
      return codeBlock;
});
```
And then change
```handlebars
<pre><code id="code-block-{{@../index}}-{{@index}}">{{line.code}}</code></pre>
```
to (something similar to)
```handlebars
<pre><code id="code-block-{{@../index}}-{{@index}}">{{codeBlock line.code}}</code></pre>
```

I appreciate your time! This extension fulfills a great need when trying to provide directed feedback to peers and students alike. I'll keep an eye on the issue so feel free to let me know if you need anything from me, or if there's something I can help with.

**Comment by d-koppenhagen** [comment]
Hey, I think I  got the point and to be honest: The `<script>` tag in the generated template is more like a workaround as I didn't manage until now to get Handlebars `registerHelper` function working.
The encoding / decoding in Base64 is just for a very simple reason: I don't need to escape all source code that will be inserted. E.g. References to lines in HTML files or CSV files can cause trouble when I would put the directly into the generated CSV. That's why it's BASE64 encoded.
I like the suggestion very much and I just tried it out again as you wrote it down, but it won't work this way.

Would you mind to simply fork this repo, make the changes you suggest and create a PR for it? If you need help for running the extension locally in debug mode etc., I can help you out.

**Comment by bluenick2k17** [comment]
I sure can! I'll let you know if I end up hitting any roadblocks.

**Comment by bluenick2k17** [comment]
I ended up figuring out how to do it, I realized that I needed to use typescript friendly functions in the helper, and not javascript friendly functions. I've had a bit of weirdness trying to run the extension tests but I'll finish up and get a pull request over to you so you can take a look.

**Comment by bluenick2k17** [comment]
I just raised the following Pull Request: #36 
I didn't end up adding the tests as I wasn't able to get them run yet on my local machine to my knowledge, but if you see that additional tests need to be written please feel free to let me know.

**Comment by d-koppenhagen** [comment]
Just released version 1.11.0. Please give it a try. It should work as expected now. Thanks for contributing.

**Comment by bluenick2k17** [comment]
It sure does work, thanks for the quick turnaround on this!
